core-bundle/src/Routing/ResponseContext/JsonLd/ContaoPageSchema.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Routing\ResponseContext\JsonLd;
+
+use Contao\CoreBundle\Routing\ResponseContext\HtmlHeadBag\HtmlHeadBag;
+use Spatie\SchemaOrg\BaseType;
+
+class ContaoPageSchema extends BaseType
+{
+    /**
+     * @var string
+     */
+    private $title;
+
+    /**
+     * @var int
+     */
+    private $pageId;
+
+    /**
+     * @var bool
+     */
+    private $noSearch;
+
+    /**
+     * @var bool
+     */
+    private $protected;
+
+    /**
+     * @var array<int>
+     */
+    private $groups;
+
+    /**
+     * @var bool
+     */
+    private $fePreview;
+
+    /**
+     * ContaoPageSchema constructor.
+     *
+     * @param array<int> $groups
+     */
+    public function __construct(string $title, int $pageId, bool $noSearch, bool $protected, array $groups, bool $fePreview)
```

Comment:
Can't we pass the PageModel here?

Reply:
No, because of the whole encoding/decoding stuff.